spec/UserDbSpec.js

```diff
@@ -2,20 +2,47 @@
 /* global it */
 /* global expect */
 
-const UserDB = require("../src/backend/UserDb.js");
+/* Imports for test */
+const UserDb = require("../src/backend/UserDb.js");
 
 /* Write the test suite */
-describe("User Db", () => {
+describe("Usser DB", () => {
     /* A setup function
-       beforeEach(() => {
-       });
-       */
+    beforeEach(() => {
+    });
+    */
+
+    /* A specification */
+    it("shall be able to retrieve a user with a correct username and password", async function() {
+        let res = await UserDb.addUser("vcedgar", "1234");
+        expect(res).toBe(true);
+
+        let result1 = await UserDb.userLogin("vcedgar", "1234");
+        let result2 = await UserDb.userLogin("vcedgar", "12345");
+        let result3 = await UserDb.userLogin("vcedgar6", "1234");
+        expect(result1).toBe(true);
+        expect(result2).toBe(false);
+        expect(result3).toBe(false);
+    });
 
     /* A specification */
-    it("shall be able to create a default user with role: student", async function() {
-        const userDb = UserDB;
-        userDb.addUser("test", "test");
-        const foundUser = await userDb.userExists("test", null);
-        expect(foundUser).toBe(true)
+    it("shall be able to add a user with a new username and fail at adding a user with an existing username", async function() {
```

Comment:
You shouldn't have "and"s in specs

Reply:
This means I will have to copy and paste nearly the exact same test because they are related :/

Reply:
Just changed the spec name then. Something like "shall add new users only when their username does not already exist"

Reply:
Done